core/src/banking_stage/transaction_scheduler/prio_graph_scheduler.rs

```diff
@@ -154,62 +156,54 @@ impl PrioGraphScheduler {
-                if !pre_lock_filter(transaction) {
-                    container.remove_by_id(&id.id);
-                    continue;
-                }
-
-                // Check if this transaction conflicts with any blocked transactions
-                if !blocking_locks.check_locks(transaction.message()) {
-                    blocking_locks.take_locks(transaction.message());
-                    unschedulable_ids.push(id);
-                    saturating_add_assign!(num_unschedulable, 1);
-                    continue;
-                }
-
-                // Schedule the transaction if it can be.
-                let transaction_locks = transaction.get_account_locks_unchecked();
-                let Some(thread_id) = self.account_locks.try_lock_accounts(
-                    transaction_locks.writable.into_iter(),
-                    transaction_locks.readonly.into_iter(),
-                    ThreadSet::any(num_threads),
+                let maybe_schedule_info = try_schedule_transaction(
+                    transaction_state,
+                    &pre_lock_filter,
+                    &mut blocking_locks,
+                    &mut self.account_locks,
+                    num_threads,
                     |thread_set| {
                         Self::select_thread(
                             thread_set,
                             &batches.transactions,
                             self.in_flight_tracker.num_in_flight_per_thread(),
                         )
                     },
-                ) else {
-                    blocking_locks.take_locks(transaction.message());
-                    unschedulable_ids.push(id);
-                    saturating_add_assign!(num_unschedulable, 1);
-                    continue;
-                };
-
-                saturating_add_assign!(num_scheduled, 1);
-
-                let sanitized_transaction_ttl = transaction_state.transition_to_pending();
-                let cost = transaction_state.cost();
+                );
 
-                let SanitizedTransactionTTL {
-                    transaction,
-                    max_age_slot,
-                } = sanitized_transaction_ttl;
-
-                batches.transactions[thread_id].push(transaction);
-                batches.ids[thread_id].push(id.id);
-                batches.max_age_slots[thread_id].push(max_age_slot);
-                saturating_add_assign!(batches.total_cus[thread_id], cost);
-
-                // If target batch size is reached, send only this batch.
-                if batches.ids[thread_id].len() >= TARGET_NUM_TRANSACTIONS_PER_BATCH {
-                    saturating_add_assign!(num_sent, self.send_batch(&mut batches, thread_id)?);
-                }
+                match maybe_schedule_info {
+                    Err(TransactionSchedulingError::Filtered) => {
+                        container.remove_by_id(&id.id);
+                        continue;
```

Comment:
i think this `continue` is needless like the `Err(TransactionSchedulingError::UnschedulableConflicts)` arm?

Reply:
e644ad978d